### DISCUSSION TITLE
The ability to control clipboard access.

### DISCUSSION THREAD

**Discussion by qvipin** [zen-browser/desktop#3869] [Ideas]
This feature would be amazing, as it’s currently lacking in Firefox but already available in Chrome. It’s an essential part of my workflow, especially when using code-server, where I frequently need to paste text.

My current solution is to set dom.events.testing.asyncClipboard to true. However, this approach is risky, as enabling this setting poses potential security concerns. Despite the danger, I rely on it due to the necessity of frequent copy-pasting in my workflow.

Having this as a built-in feature or a user-configurable permission in Firefox would be a significant improvement. Since Chrome already supports this functionality, implementing it in Firefox should be feasible. It would be a blessing for users like me who depend heavily on clipboard operations for development tasks.

**Comment by vraravam** [comment]
Couldn't this be abused by malware to capture keystrokes? (asking out of curiosity)

**Reply by qvipin** [reply]
Yeah that is why I feel very compromised setting dom.events.testing.asyncClipboard to true
